### ISSUE TITLE
a11y: Remove filter control is not keyboard focusable/actionable

### ISSUE THREAD

**Issue by iamjolly** [18F/all_sorns#149]
## Description:
The remove filter (x) control is an anchor element that is missing an `href` attribute and value. This makes the link inert/placeholder in browsers and is not keyboard focusable or actionable. This is the same issue as #136, but in a slightly different form.

## Screenshot:
![Screen Shot 2021-01-22 at 5 02 33 PM](https://user-images.githubusercontent.com/1093423/105561369-c81b0180-5cd3-11eb-8412-01f27baea1cc.png)

## WCAG SC:
[Success Criterion 2.1.1 Keyboard (Level A)](https://www.w3.org/WAI/WCAG21/Understanding/keyboard.html): All functionality of the content is operable through a keyboard interface without requiring specific timings for individual keystrokes, except where the underlying function requires input that depends on the path of the user's movement and not just the endpoints.

## Recommended fix:
Use a `button` element, if possible or [`input type="image"`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/image) with the appropriate `alt` text to indicate "remove this filter" (not "close").
